libsolidity/ast/Types.cpp

```diff
@@ -340,11 +368,138 @@ MemberList::MemberMap IntegerType::nativeMembers(ContractDefinition const*) cons
 		return MemberList::MemberMap();
 }
 
-bool IntegerConstantType::isValidLiteral(const Literal& _literal)
+FixedPointType::FixedPointType(int _integerBits, int _fractionalBits, FixedPointType::Modifier _modifier):
+	m_integerBits(_integerBits), m_fractionalBits(_fractionalBits), m_modifier(_modifier)
+{
+	solAssert(
+		m_integerBits + m_fractionalBits > 0 && 
+		m_integerBits + m_fractionalBits <= 256 && 
+		m_integerBits % 8 == 0 && 
+		m_fractionalBits % 8 == 0,
+		"Invalid bit number(s) for fixed type: " + 
```

Comment:
I think we have to add another restriction here: If the fixed point type is signed, the integer part cannot have zero bits, because we need at least one bit for the sign.
An assert should be added here, and Token.cpp should not generate/parse the tokens `fixed0x8`, ..., `fixed0x256` in the first place.